### DISCUSSION TITLE
Combining Sanity with Supabase

### DISCUSSION THREAD

**Discussion by chrcit** [supabase/supabase#1438] [Questions] [answered]
Hello, 
my name is Christian and I'm from Vienna, Austria 👋 

I'm currently building a webapp for a shortfilm festival. We're using Sanity as a CMS for the content (programs, films, filmmaker profiles) and want to use Supabase to add user authentication as well as some real-time functionality to give the feel of a live festival.
After the festival we're planning to release all the code as an open source project.

The plan is to add live watchers to each film (just a small indicator) and a live chat with an Instagram like applause feature.

To do this we would need to add the some sanity id's to the Supabase tables as external ids. Does anything speak against interoperating the two systems like this? Would love your input on it :)

Also: Thank you so much for creating Supabase. We were looking into Firebase and were really dreading giving our business and data to Google.

Best Regards,
Christian

**Comment by inian** [comment, marked as answer]
Hey Christian, 

That sounds pretty cool! 

Our realtime engine captures changes in database state (when new rows are added, updated, etc). It looks like you would need something like "presence" for the number of live watchers. We have some ideas around [this](https://github.com/supabase/supabase/discussions/257#discussioncomment-108284) but nothing concrete yet. That being said, you can probably do something like insert a new row into the database everytime someone visits or leaves the website, but do benchmark it to see how this scales. 

Apart from this, I don't see any other blockers with Sanity etc - Supabase is raw Postgres, so you can store anything there :) 

Looking forward to seeing what you build.

**Reply by fergusmeiklejohn** [reply]
I think Presence would be a really important feature Supabase could have while Firestore still does not offer it.  So many apps are about team work and networking, it's very useful to know who is in the virtual room.

**Reply by chrcit** [reply]
Thanks for your reply!

I was thinking about inserting a new row in the film <> viewer relationship table each time a viewer has a specific movie open for longer than x seconds.

The goal would be to display the specific user avatars as icons.

Is it possible to listen to a specific table (`film_viewer`) and have the payload include all new rows joined with the `viewer` table?

If that doesn't work (or we have performance issues) we would add a `live_viewer_count` column to the `film` table and listen to updates on that. Then just increment/decrement on the client.

I agree having presence in Supabase would be amazing!

Y'all are doing great work, I'm really looking forward to seeing the future of Supabase!

We're just 2 people working on this project on the side.
Supabase + Sanity + Next.js/Vercel allow us to 10x our productivity while also having 10x the DX.

We'll definitely do a writeup of the stack when we open-source the project in the summer.

**Reply by inian** [reply]
>  inserting a new row in the film <> viewer relationship table each time a viewer has a specific movie open for longer than x seconds.

Yeah, that's the way I would do it as well for now. 

> Is it possible to listen to a specific table (film_viewer) and have the payload include all new rows joined with the viewer table?

Right now realtime server only returns the row that has been changed. You will need to make another API call to get the other records but that wont be efficient. Its be better to have the data you are looking for in the same table as you mentioned. 

> Supabase + Sanity + Next.js/Vercel allow us to 10x our productivity while also having 10x the DX.
We'll definitely do a writeup of the stack when we open-source the project in the summer.

🎉

**Reply by nk9** [reply]
I'm interested in this stack, @chrcit. Did you ever open source your project?

**Reply by chrcit** [reply]
Hi @nk9, no I never did. The client changed her mind about wanting to open source it.

If I were to use this stack today I'd use GraphQL to combine Sanity and Supabase into one schema.
